src/metabase/util/ssh.clj

```diff
@@ -1,32 +1,63 @@
 (ns metabase.util.ssh
-  (:require [clojure.string :as str]
-            [clojure.tools.logging :as log]
+  (:require [clojure.tools.logging :as log]
             [metabase.util :as u])
-  (:import com.jcraft.jsch.JSch))
+  (:import java.io.ByteArrayInputStream
+           [org.apache.sshd.client.future AuthFuture ConnectFuture]
+           org.apache.sshd.client.session.ClientSession
+           org.apache.sshd.client.session.forward.PortForwardingTracker
+           org.apache.sshd.client.SshClient
+           [org.apache.sshd.common.config.keys FilePasswordProvider FilePasswordProvider$ResourceDecodeResult]
+           org.apache.sshd.common.session.SessionHolder
+           org.apache.sshd.common.util.GenericUtils
+           org.apache.sshd.common.util.io.resource.AbstractIoResource
+           org.apache.sshd.common.util.net.SshdSocketAddress
+           org.apache.sshd.common.util.security.SecurityUtils
+           org.apache.sshd.server.forward.AcceptAllForwardingFilter))
 
 (def ^:private default-ssh-timeout 30000)
 
+(def ^:private client
+  (doto (SshClient/setUpDefaultClient)
+    (.start)
+    (.setForwardingFilter AcceptAllForwardingFilter/INSTANCE)))
+
 (defn start-ssh-tunnel
```

Comment:
There are varying opinions on which functions should end in `!` in the Clojure world but generally we've been following the "functions that change state should end in `!`" -- https://guide.clojure.style/#changing-state-fns-with-exclamation-mark

IMO `start-ssh-tunnel!` and `close-tunnel!` should end in `!` because calling them once is very different from calling them 20 times (which *could* happen inside an STM transaction)